GVFS/GVFS.Common/LocalCacheResolver.cs

```diff
@@ -56,7 +56,11 @@ public static bool TryGetDefaultLocalCacheRoot(GVFSEnlistment enlistment, out st
             {
                 // A lock is required because FileBasedDictionary is not multi-process safe, neither is the act of adding a new cache
                 string lockPath = Path.Combine(localCacheRoot, MappingFile + ".lock");
-                this.fileSystem.CreateDirectory(localCacheRoot);
+
+                // Create any missing parent directories of localCacheRoot with default permissions/ACLs
+                // before creating localCacheRoot itself
+                this.fileSystem.CreateDirectory(Path.GetDirectoryName(localCacheRoot));

```

Comment:
Aren't we already doing this inside of `CreateDirectoryAccessibleByAuthUsers`?
We should make clear what the responsibilities are of `CreateDirectoryAccessibleByAuthUsers` and what preconditions/postconditions are.

Reply:
Thanks for pointing this out!

I went back and forth on who would be responsible for creating the intermediary directories, and I missed removing the code from the callers.

Initially the changes in this PR were much larger in scope (and included the changes for #1579) but after seeing those changes get too large/complex I scaled them back and missed cleaning up this code.